asusd/src/ctrl_slash/mod.rs

```diff
@@ -0,0 +1,102 @@
+pub mod config;
+pub mod trait_impls;
+
+use rog_platform::hid_raw::HidRaw;
+use rog_platform::usb_raw::USBRaw;
+use rog_slash::{SlashMode, SlashType};
+use rog_slash::error::SlashError;
+use rog_slash::usb::{get_slash_type, pkt_set_mode, pkt_set_options, pkts_for_init};
+use crate::ctrl_slash::config::SlashConfig;
+use crate::error::RogError;
+
+enum Node {
+    Usb(USBRaw),
+    Hid(HidRaw),
+}
+
+impl Node {
+    pub fn write_bytes(&self, message: &[u8]) -> Result<(), RogError> {
+        // TODO: map and pass on errors
+        match self {
+            Node::Usb(u) => {
+                u.write_bytes(message).ok();
+            }
+            Node::Hid(h) => {
+                h.write_bytes(message).ok();
+            }
+        }
+        Ok(())
+    }
+}
+
+pub struct CtrlSlash {
+    // node: HidRaw,
+    node: Node,
+    config: SlashConfig,
+    // slash_type: SlashType,
```

Comment:
There is the G16 version, if you can please add handling for it and we will assume it is the same length for now.